piptools/scripts/sync.py

```diff
@@ -108,6 +119,26 @@ def cli(
             log.error("ERROR: " + msg)
             sys.exit(2)
 
+    if python_executable:
+        resolved_python_executable = shutil.which(python_executable)
+        if resolved_python_executable is None:
+            msg = "Could not resolve '{}' as valid executable path or alias"
+            log.error(msg.format(python_executable))
+            sys.exit(2)
+
+        # Ensure that target python executable has the right version of pip installed
+        pip_version = get_pip_version_for_python_executable(python_executable)
+        required_pip_specification = get_required_pip_specification()
+        if not required_pip_specification.contains(pip_version):
+            msg = (
+                "Target python executable '{}' has pip version {} installed. "
+                "Version {} is expected."
+            )
+            log.error(
+                msg.format(python_executable, pip_version, required_pip_specification)
```

Comment:
FWIW it's recommended (although controversial sometimes) to let the logging module take care of rendering things. And this is why it's best to pass a separate string with percent-placeholders as the first arg and pass the vars separately, instead of pre-rendering that string.
This also allows different tooling to aggregate log entries coming from the same place but with different args.

Reply:
Thanks, I updated logging calls. 

BTW, I was just following the pattern used in the file, it could be a separate discussion if all log statements should be aligned and switch to logging formatting.